### PR TITLE
 Steps form: Automatically converts a form into steps. 

### PR DESCRIPTION
**Pull request by GormFrank**

### CODE REVIEW

File: src/plugins/stepsform/stepsform-en.hbs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+							<label for="defNot-2"><input type="radio" name="ethics2" value="defNot-2" id="defNot-2"/>&#160;Definitely Not</label>&#160;
+							<label for="probNot-2"><input type="radio" name="ethics2" value="probNot-2" id="probNot-2"/>&#160;Probably Not</label>&#160;
+							<label for="maybe-2"><input type="radio" name="ethics2" value="maybe-2" id="maybe-2"/>&#160;Maybe (Not Sure)</label>&#160;
+							<label for="probably-2"><input type="radio" name="ethics2" value="probably-2" id="probably-2"/>&#160;Probably</label>&#160;
+							<label for="definitely-2"><input type="radio" name="ethics2" value="definitely-2" id="definitely-2"/>&#160;Definitely</label>
+						</div>
+					</fieldset>
+				</div>
+			</fieldset>
+			<input type="submit" name="submit" value="Submit" class="btn btn-primary" />
```
Comment (duboisp) [inline src/plugins/stepsform/stepsform-en.hbs]: For some reason, after the plugin initialization that input is moved inside the preceding fieldset which is are not totally right from a sementic point of view.

Can you check to move all the navigation button outside the fieldset?

File: src/plugins/stepsform/stepsform.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+			/*
+			 * Variable and function definitions
+			 * These will be initialized once per instance of plugin.
+			 */
+			var form = elm.getElementsByTagName( "FORM" )[ 0 ],
+				fieldsets = ( form ) ? $( form ).children( "fieldset" ) : 0,
+				hasStepsInitialized;
+
+			// Initialize navigation buttons
+			btnPrevious = createStepsButton( "A", "prev", "mrgn-rght-sm mrgn-bttm-md", i18nText.prv );
```
Comment (duboisp) [inline src/plugins/stepsform/stepsform.js]: Instead of using anchor, can you change it for button?

**Review by duboisp — CHANGES_REQUESTED** [review]

File: src/plugins/stepsform/stepsform.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+	 */
+	createStepsButton = function( tagName, type, style, text ) {
+		var control = document.createElement( tagName ),
+			btnPrimary = "btn btn-md btn-primary",
+			btnDefault = "btn btn-md btn-default";
+
+		// set default attributes
+		control.className = ( type === "prev" ? btnDefault : btnPrimary ) + " " + style;
+		control.href = "#";
+		control.setAttribute( "aria-labelby", text );
```
Comment (duboisp) [inline src/plugins/stepsform/stepsform.js]: Remove the aria-labelby as it don't contain a valid ID

File: src/plugins/stepsform/stepsform.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+	 * @returns {Object} A ready-to-use button element
+	 */
+	createStepsButton = function( tagName, type, style, text ) {
+		var control = document.createElement( tagName ),
+			btnPrimary = "btn btn-md btn-primary",
+			btnDefault = "btn btn-md btn-default";
+
+		// set default attributes
+		control.className = ( type === "prev" ? btnDefault : btnPrimary ) + " " + style;
+		control.href = "#";
```
Comment (duboisp) [inline src/plugins/stepsform/stepsform.js]: This is not needed with a button

**Review by duboisp — CHANGES_REQUESTED** [review]

File: site/pages/docs/ref/stepsform/stepsform-en.hbs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	<ul>
+		<li>Web forms are long/contain lot of information.</li>
+	</ul>
+</section>
+
+<section>
+	<h2>Working example</h2>
+	<ul>
+		<li><a href="../../../demos/stepsform/stepsform-en.html">English example</a></li>
+		<li><a href="../../../demos/stepsform/stepsform-fr.html">French example</a></li>
```
[the remark this answers is not in the document]
Comment (namjohn920) [reply site/pages/docs/ref/stepsform/stepsform-en.hbs]: resolved

File: site/pages/docs/ref/stepsform/stepsform-fr.hbs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+	<h2>Utiliser lorsque</h2>
+	<ul>
+		<li>Les formulaires sont longs ou contiennent un substantiel d'information.</li>
+	</ul>
+</section>
+
+<section>
+	<h2>Exemple pratique</h2>
+	<ul>
+		<li><a href="../../../demos/stepsform/stepsform-en.html">Exemple en anglais</a></li>
```
[the remark this answers is not in the document]
Comment (namjohn920) [reply site/pages/docs/ref/stepsform/stepsform-fr.hbs]: resolved

File: src/plugins/stepsform/stepsform-en.hbs [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+					<div class="form-group">
+						<label for="email"><span class="field-name">Email Address</span> (test@domain)</label>
+						<input class="form-control" id="email" name="email" type="email" />
+					</div>
+				</div>
+			</fieldset>
+			<fieldset id="questionaire">
+				<legend>Ethics Questionnaire</legend>
+				<div>
+					<small>For each of the following examples, indicate whether or not you feel that there is an ethical issue.</small>
```
[the remark this answers is not in the document]
Comment (namjohn920) [reply src/plugins/stepsform/stepsform-en.hbs]: resolved

File: src/plugins/stepsform/stepsform-en.hbs [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+					<fieldset style="border:0;">
```
[the remark this answers is not in the document]
Comment (namjohn920) [reply src/plugins/stepsform/stepsform-en.hbs]: resolved

File: src/plugins/stepsform/stepsform-en.hbs [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+						<legend><span class="field-name">Ethics 1</span>: You pay $10.00 bucks for a purchase, but receive change for $20.00 bucks. Would you keep the extra money?</legend>
+						<div class="radio">
+							<label for="defNot-1"><input type="radio" name="ethics1" value="defNot-1" id="defNot-1"/>&#160;Definitely Not</label>&#160;
+							<label for="probNot-1"><input type="radio" name="ethics1" value="probNot-1" id="probNot-1"/>&#160;Probably Not</label>&#160;
+							<label for="maybe-1"><input type="radio" name="ethics1" value="maybe-1" id="maybe-1"/>&#160;Maybe (Not Sure)</label>&#160;
+							<label for="probably-1"><input type="radio" name="ethics1" value="probably-1" id="probably-1"/>&#160;Probably</label>&#160;
+							<label for="definitely-1"><input type="radio" name="ethics1" value="definitely-1" id="definitely-1"/>&#160;Definitely</label>
+						</div>
+					</fieldset>
+					<fieldset style="border:0;">
```
[the remark this answers is not in the document]
Comment (namjohn920) [reply src/plugins/stepsform/stepsform-en.hbs]: resolved

File: src/plugins/stepsform/stepsform-en.hbs (hunk 1 above)
Comment (namjohn920) [reply src/plugins/stepsform/stepsform-en.hbs]: resolved

File: src/plugins/stepsform/stepsform.js (hunk 2 above)
Comment (namjohn920) [reply src/plugins/stepsform/stepsform.js]: resolved

File: src/plugins/stepsform/stepsform.js (hunk 4 above)
Comment (namjohn920) [reply src/plugins/stepsform/stepsform.js]: resolved

File: src/plugins/stepsform/stepsform.js (hunk 3 above)
Comment (namjohn920) [reply src/plugins/stepsform/stepsform.js]: resolved

**Comment by duboisp** [comment]
Completed via #8811

**Pull request closed without merging**